File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/PypiClusterLibrarySummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details of a PyPI package installed as a library on a cluster.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=PypiClusterLibrarySummary.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="type")

public final class PypiClusterLibrarySummary extends ClusterLibrarySummary {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("installedBy")
private String installedBy;

public Builder installedBy(String installedBy) {
    this.installedBy = installedBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("installedByName")
private String installedByName;

public Builder installedByName(String installedByName) {
    this.installedByName = installedByName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("stateMessage")
private String stateMessage;

public Builder stateMessage(String stateMessage) {
    this.stateMessage = stateMessage;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("status")
private Status status;

public Builder status(Status status) {
    this.status = status;
    return this;
}
            /**
     * PyPI package specifier, for example {@code requests==2.31.0}.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("pythonPackage")
private String pythonPackage;

        /**
         * PyPI package specifier, for example {@code requests==2.31.0}.
         * @param pythonPackage the value to set
         * @return this builder
         **/
        

public Builder pythonPackage(String pythonPackage) {
    this.pythonPackage = pythonPackage;
    return this;
}
            /**
     * Optional pip index URL used to resolve the package, for example {@code https://pypi.org/simple}.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("pipIndexUrl")
private String pipIndexUrl;

        /**
         * Optional pip index URL used to resolve the package, for example {@code https://pypi.org/simple}.
         * @param pipIndexUrl the value to set
         * @return this builder
         **/
        

public Builder pipIndexUrl(String pipIndexUrl) {
    this.pipIndexUrl = pipIndexUrl;
    return this;
}


        public PypiClusterLibrarySummary build() {
            PypiClusterLibrarySummary model = new PypiClusterLibrarySummary(this.timeCreated
                , this.timeUpdated
                , this.installedBy
                , this.installedByName
                , this.stateMessage
                , this.status
                , this.pythonPackage
                , this.pipIndexUrl);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(PypiClusterLibrarySummary model) {
                this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.installedBy(model.getInstalledBy());
    this.installedByName(model.getInstalledByName());
    this.stateMessage(model.getStateMessage());
    this.status(model.getStatus());
    this.pythonPackage(model.getPythonPackage());
    this.pipIndexUrl(model.getPipIndexUrl());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public PypiClusterLibrarySummary(java.util.Date timeCreated, java.util.Date timeUpdated, String installedBy, String installedByName, String stateMessage, Status status, String pythonPackage, String pipIndexUrl) {
    super(timeCreated, timeUpdated, installedBy, installedByName, stateMessage, status);
        this.pythonPackage = pythonPackage;
        this.pipIndexUrl = pipIndexUrl;
    }


        /**
     * PyPI package specifier, for example {@code requests==2.31.0}.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("pythonPackage")
    private final String pythonPackage;

        /**
     * PyPI package specifier, for example {@code requests==2.31.0}.
     * @return the value
     **/
    
    public String getPythonPackage() {
        return pythonPackage;
    }


        /**
     * Optional pip index URL used to resolve the package, for example {@code https://pypi.org/simple}.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("pipIndexUrl")
    private final String pipIndexUrl;

        /**
     * Optional pip index URL used to resolve the package, for example {@code https://pypi.org/simple}.
     * @return the value
     **/
    
    public String getPipIndexUrl() {
        return pipIndexUrl;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("PypiClusterLibrarySummary(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", pythonPackage=").append(String.valueOf(this.pythonPackage));
        sb.append(", pipIndexUrl=").append(String.valueOf(this.pipIndexUrl));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PypiClusterLibrarySummary)) {
            return false;
        }

        PypiClusterLibrarySummary other = (PypiClusterLibrarySummary) o;
        return java.util.Objects.equals(this.pythonPackage, other.pythonPackage) &&
            java.util.Objects.equals(this.pipIndexUrl, other.pipIndexUrl) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.pythonPackage == null ? 43 : this.pythonPackage.hashCode());
        result = (result * PRIME) + (this.pipIndexUrl == null ? 43 : this.pipIndexUrl.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/pypi_cluster_library_summary.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.


from .cluster_library_summary import ClusterLibrarySummary
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class PypiClusterLibrarySummary(ClusterLibrarySummary):
    """
    Details of a PyPI package installed as a library on a cluster.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new PypiClusterLibrarySummary object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.PypiClusterLibrarySummary.type` attribute
        of this class is ``PYPI`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param type:
            The value to assign to the type property of this PypiClusterLibrarySummary.
        :type type: str

        :param time_created:
            The value to assign to the time_created property of this PypiClusterLibrarySummary.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this PypiClusterLibrarySummary.
        :type time_updated: datetime

        :param installed_by:
            The value to assign to the installed_by property of this PypiClusterLibrarySummary.
        :type installed_by: str

        :param installed_by_name:
            The value to assign to the installed_by_name property of this PypiClusterLibrarySummary.
        :type installed_by_name: str

        :param state_message:
            The value to assign to the state_message property of this PypiClusterLibrarySummary.
        :type state_message: str

        :param status:
            The value to assign to the status property of this PypiClusterLibrarySummary.
            Allowed values for this property are: "PENDING", "RESOLVING", "INSTALLING", "INSTALLED", "FAILED", "INSTALL_ON_RESTART", "UNINSTALL_ON_RESTART", "SKIPPED", "DELETED"
        :type status: str

        :param python_package:
            The value to assign to the python_package property of this PypiClusterLibrarySummary.
        :type python_package: str

        :param pip_index_url:
            The value to assign to the pip_index_url property of this PypiClusterLibrarySummary.
        :type pip_index_url: str

        """
        self.swagger_types = {
            'type': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'installed_by': 'str',
            'installed_by_name': 'str',
            'state_message': 'str',
            'status': 'str',
            'python_package': 'str',
            'pip_index_url': 'str'
        }

        self.attribute_map = {
            'type': 'type',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'installed_by': 'installedBy',
            'installed_by_name': 'installedByName',
            'state_message': 'stateMessage',
            'status': 'status',
            'python_package': 'pythonPackage',
            'pip_index_url': 'pipIndexUrl'
        }

        self._type = None
        self._time_created = None
        self._time_updated = None
        self._installed_by = None
        self._installed_by_name = None
        self._state_message = None
        self._status = None
        self._python_package = None
        self._pip_index_url = None
        self._type = 'PYPI'

    @property
    def python_package(self):
        """
        Gets the python_package of this PypiClusterLibrarySummary.
        PyPI package specifier, for example `requests==2.31.0`.


        :return: The python_package of this PypiClusterLibrarySummary.
        :rtype: str
        """
        return self._python_package

    @python_package.setter
    def python_package(self, python_package):
        """
        Sets the python_package of this PypiClusterLibrarySummary.
        PyPI package specifier, for example `requests==2.31.0`.


        :param python_package: The python_package of this PypiClusterLibrarySummary.
        :type: str
        """
        self._python_package = python_package

    @property
    def pip_index_url(self):
        """
        Gets the pip_index_url of this PypiClusterLibrarySummary.
        Optional pip index URL used to resolve the package, for example `https://pypi.org/simple`.


        :return: The pip_index_url of this PypiClusterLibrarySummary.
        :rtype: str
        """
        return self._pip_index_url

    @pip_index_url.setter
    def pip_index_url(self, pip_index_url):
        """
        Sets the pip_index_url of this PypiClusterLibrarySummary.
        Optional pip index URL used to resolve the package, for example `https://pypi.org/simple`.


        :param pip_index_url: The pip_index_url of this PypiClusterLibrarySummary.
        :type: str
        """
        self._pip_index_url = pip_index_url

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
